### PR TITLE
Do not expose return_predictions in LightningCLI predict subcommand

### PR DESCRIPTION
**Pull request by mauvilsa**
## What does this PR do?

For the `predict` subcommand, the `return_predictions` parameter is now skipped. From what I understand, this parameter makes no sense to be an argument of the CLI.

Fixes #10509

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/cli.py [hunk 1]
```diff
@@ -633,7 +633,7 @@ def subcommands() -> Dict[str, Set[str]]:
             "fit": {"model", "train_dataloaders", "val_dataloaders", "datamodule"},
             "validate": {"model", "dataloaders", "datamodule"},
             "test": {"model", "dataloaders", "datamodule"},
-            "predict": {"model", "dataloaders", "datamodule"},
+            "predict": {"model", "dataloaders", "datamodule", "return_predictions"},
```
Comment (carmocca) [inline pytorch_lightning/utilities/cli.py:636]: Why does it not make sense?

It could be useful with `LightningCLI(run=False)` or a custom CLI that overrides `LightningCLI.predict`

Comment (mauvilsa) [reply pytorch_lightning/utilities/cli.py:636]: `run=False` disables subcommands and is not affected by this change. From what I understand `return_predictions` affects the return value of `Trainer.predict`. And this return value is ignored in the code, so it does nothing whatever the value:

https://github.com/PyTorchLightning/pytorch-lightning/blob/d132a9c3b73bad3e893f809a53401722087bec5e/pytorch_lightning/utilities/cli.py#L823

Maybe I am wrong and not understanding how it works.

Comment (carmocca) [reply pytorch_lightning/utilities/cli.py:636]: This works:

```python
import torch
from torch.utils.data import Dataset, DataLoader

from pytorch_lightning import LightningModule
from pytorch_lightning.utilities.cli import LightningCLI


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()

    def predict_step(self, batch, batch_idx: int, dataloader_idx: int = 0):
        return 123

    def predict_dataloader(self):
        return DataLoader(RandomDataset(32, 64), batch_size=2)


class MyLightningCLI(LightningCLI):
    def predict(self, **kwargs):
        predictions = self.trainer.predict(**kwargs)
        print(predictions)


if __name__ == "__main__":
    MyLightningCLI(BoringModel)
```

```python
$ python script.py predict --return_predictions=True
```

Comment (carmocca) [reply pytorch_lightning/utilities/cli.py:636]: I agree this should be included in the documentation. Would you like to use this PR for it?

`return_predictions` is `True` by default which is useless with the CLI unless you do the above. But I wouldn't have the CLI changing it.

Comment (mauvilsa) [reply pytorch_lightning/utilities/cli.py:636]: But I still don't understand what is the benefit of exposing it. When would someone ever want to do `--return_predictions=False`? If I run predict is because I want the predictions. The only other use case I can think of is I want to know how long the predictions take but don't care about the output. But that can be achieved by redirecting to `/dev/null`. On the other hand, keeping it exposed creates confusion.

Comment (rohitgr7) [reply pytorch_lightning/utilities/cli.py:636]: > When would someone ever want to do `--return_predictions=False`

possibly saving predictions to the disk rather than returning them due to memory overhead?

Comment (mauvilsa) [reply pytorch_lightning/utilities/cli.py:636]: Saving to disk would mean that the `predict` method is implemented and there you save to a file. But this requires `return_predictions=True`, no?

Comment (rohitgr7) [reply pytorch_lightning/utilities/cli.py:636]: I can save inside `predict_step` itself to avoid accumulating predictions for the whole dataset and just set `return_predictions=False`

```py
def predict_step(...):
    preds = ...
    # save preds
```

Comment (mauvilsa) [reply pytorch_lightning/utilities/cli.py:636]: Okay, now I think I understand the entire picture. Anyway I think some things can be done to support all use cases and be more intuitive. Also the method `LightningCLI.predict` imposes a limitation so I would have thought of doing it a bit different. I would prefer to not document this just yet. For the moment I will close this pull request and will come back with a proposal in an new issue.
